shimmingtoolbox/optimizer/lsq_optimizer.py

```diff
@@ -271,7 +273,9 @@ def optimize(self, mask):
 
         # Set up output currents
         currents_0 = self.get_initial_guess()
-
+        #In case if there is nothing to shimmed
+        if np.all(unshimmed_vec == 0):
+            return currents_0
```

Comment:
I don't think you shout base yourself of `currents_0`. It depends on the initial guessing method (it's 0s for now but it could also be set to return halfway through the bounds or return the values of an adjacent slice). If `unshimmed_vec` is 0s then the output currents should always be 0s.

Reply:
It's fixed !